helper/resource/testing_new.go

```diff
@@ -254,7 +255,10 @@ func runNewTest(ctx context.Context, t testing.T, c TestCase, helper *plugintest
 
 			testStepConfig = teststep.Configuration(confRequest)
 
-			err = wd.SetConfig(ctx, testStepConfig, step.ConfigVariables)
+			if !step.Query {
```

Comment:
For purposes of this PR, `SetConfig` and `SetQuery` are 2 different things because one writes/manages a `.tf` file and one writes/manages a `.tfquery.hcl` file.

Does `if !step.Query` feel right here? Likely not. The goal of this PR is more make-it-work than make-it-right/make-it-well-factored 😃